### PR TITLE
Jvenstad/more general roles

### PR DESCRIPTION
**Pull request by jonmv**
@mpolden and @tokle please review. This is a first take on our general roles.

### CODE REVIEW

**Comment by jonmv** [comment]
Need more roles. We should let users determine what roles are right for them. 

Also, we need to find all GET stuff we haven’t covered yet, like the node repo, config, etc..

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/role/Policy.java [hunk 1]
```diff
@@ -12,40 +12,85 @@
...
+
+    /** Full access to application development deployments. */
+    development(Privilege.grant(Action.all())
+                         .on(PathGroup.development)
+                         .in(SystemName.all())),
+
+    /** Full access to application production deployments. */
+    production(Privilege.grant(Action.all())
+                        .on(PathGroup.deployment)
+                        .in(SystemName.all())),
```
Comment (tokle) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/role/Policy.java]: Do we want to support this in public?

**Review by tokle — COMMENTED** [review]
Looks good, I think it makes sense to add this to the policy level, but I think we should try to agree on how to name policies. The name should (as much as feasible) reflect what the policy actually means (including any restrictions).
E.g:
development -> deployApplicationDevelopment
production -> deployApplicationProduction

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/role/Policy.java (hunk 1 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/role/Policy.java]: I think so. Self-serve emergency stuff.

**Comment by jonmv** [comment]
Agreed. I'll do another round with the policy names.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/JobControllerApiHandlerHelper.java [hunk 2]
```diff
@@ -62,7 +62,7 @@
  * @see JobController
  * @see ApplicationApiHandler
  */
-class JobControllerApiHandlerHelper {
+class JobControllerApiHandlerHelper { // TODO jvenstad: Integrate into ApplicationApiHandler
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/JobControllerApiHandlerHelper.java]: Why? ApplicationApiHandler is already way too big..

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/role/PathGroup.java [hunk 3]
```diff
@@ -17,68 +17,75 @@
...
+    /** Paths used for direct deployment to production zones. */
+    deployment(Matcher.tenant,
+               Matcher.application,
+               "/application/v4/tenant/{tenant}/application/{application}/environment/prod/{*}",
+               "/application/v4/tenant/{tenant}/application/{application}/environment/test/{*}",
+               "/application/v4/tenant/{tenant}/application/{application}/environment/staging/{*}"),
+
+    /** Paths used for continuous deployment to production. */
+    submission(Matcher.tenant,
+                 Matcher.application,
```
Comment (mpolden) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/role/PathGroup.java]: Alignment.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/JobControllerApiHandlerHelper.java (hunk 2 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/JobControllerApiHandlerHelper.java]: Size in itself isn't a problem, and these paths are served by that handler. 
I guess there's no strong reason to move them, though. But I _will_ move the tests, so the API paths are properly unit tested.

**Comment by jonmv** [comment]
Will reopen against master.

**Pull request closed without merging**